### PR TITLE
feat: Introduce support fee for Statemine

### LINKED ISSUE #5130: Introduce flat fee for BSX
### Is your feature request related to a problem?

- https://github.com/kodadot/nft-gallery/pull/1967

### Describe the solution you would like

I would like to have same functionality as we have for remark

### Describe alternatives you have considered

the stuff is that this is super tricky, you need to call `api.tx.tokens.transfer` where args are `(accountId, amount, tokenId)`

token id is `1` (for BSX) or `5` (for Snek)

### Screenshots

_No response_

### Are there opened related issues?

_No response_

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

Added support fee buy and mint collection 

- [x] Closes #5130

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; 

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a718a04</samp>

Added support fee to NFT purchase transactions on Statemine and other chains. Refactored some functions to calculate and display the support fee in the `ConfirmPurchaseModal` component and the `transactionBuy` composable. Extracted the `getPercentSupportFee` function to `utils/support.ts` for reusability.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a718a04</samp>

> _Sing, O Muse, of the skillful coder who refined the codebase_
> _With cunning functions and composable logic, to handle the support fee_
> _That the platform, rich in NFTs, demands from every purchase_
> _On any chain, be it Statemine or others, with `somePercentFromTX`_

### CODE REVIEW

**Comment by hassnian** [comment]
@kodadot/qa-guild pls check

File: components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue [hunk 1]
```diff
@@ -116,12 +114,10 @@ const totalRoyalties = computed(() =>
...
 
-const totalWithRoyalties = computed(
-  () => totalNFTsPrice.value + totalRoyalties.value,
-)
+const { supportFee } = useBuySupportFee(urlPrefix, totalNFTsPrice)
 
-const supportFee = computed(() =>
-  isRemark.value ? totalNFTsPrice.value * SUPPORT_FEE_PERCENT : 0,
+const total = computed(
+  () => totalNFTsPrice.value + totalRoyalties.value + supportFee.value,
```
Comment (roiLeo) [inline components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue:120]: are we sure they are all numbers?

**Review by roiLeo — APPROVED** [review]

File: components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue (hunk 1 above)
Comment (hassnian) [reply components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue:120]: yes
CleanShot 2023-11-23 at 18 07 42@2x

the only new value is support fee 

CleanShot 2023-11-23 at 18 08 45@2x

which is also a number

**Comment by prury** [comment]
i made a transaction and paid the $0.5 fee, but where can i see that a percentage of it got deposited on the kodadot address? 
https://assethub-kusama.subscan.io/account/F3opxRUwkBj1LqjZ7DyiHCRh9Z4zVPLaVjoxfD5ddbip8mt

**Review by preschian — APPROVED** [review]

**Comment by roiLeo** [comment]
> i made a transaction and paid the $0.5 fee, but where can i see that a percentage of it got deposited on the kodadot address? https://assethub-kusama.subscan.io/tx/0xb79140cec97adbcabee27ea0c04d3fa1c753f0b6f7d1f30cfaa40ac1d8a203df

Maybe https://assethub-kusama.subscan.io/extrinsic/5851882-2 but i doesn't look like it has worked

Capture d’écran 2023-11-24 à 8 35 29 AM

**Comment by prury** [comment]
Yup, tested again, but now creating a collection(😄) and it appears that nothing got sent:
https://assethub-kusama.subscan.io/tx/0x1ef17414ef7e2bf5ec53f5ef4c83e8017cf5b513e1e76950c73de774a0ff70e5
image
kodadot DAO address above

**Comment by hassnian** [comment]
@roiLeo pls comment if any [6045661](https://github.com/kodadot/nft-gallery/pull/8200/commits/60456615d4121844fd1de8cf5ab83f3927c5107c)

**Comment by hassnian** [comment]
@prury pls try again 

note ahp will go here 

https://www.subscan.io/account/14SprTXXg5Bdgj8qV6ds57mevDwAib4SKLg6R8Lc6TzeDcLa -> https://assethub-polkadot.subscan.io/account/14SprTXXg5Bdgj8qV6ds57mevDwAib4SKLg6R8Lc6TzeDcLa?tab=transfer


 as per code only KSM will go here 👇,  DOT will go to @KodaDot_Guild_Rewards 👆

ahk https://www.subscan.io/account/CykZSc3szpVd95PmmJ45wE4ez7Vj3xkhRFS9H4U1WdrkaFY -> https://assethub-kusama.subscan.io/account/CykZSc3szpVd95PmmJ45wE4ez7Vj3xkhRFS9H4U1WdrkaFY?tab=transfer

**Comment by prury** [comment]
collection creation on Kusama Assethub: ✔️ 
https://assethub-kusama.subscan.io/tx/0x28e3463b73f8c8f4f824d483ae78771d0eae04e83b55cbfd1d47b38ba1e5d475
image

NFT creation on Kusama Assethub: 💸 
https://assethub-kusama.subscan.io/tx/0x20a0e15b2b49b404e71fe091711e9dde22ccd7e26abf4cde8ff875136536495a
image

it is going to the correct account, but it charged me $2,36

Polkadot Assethub:

**Comment by hassnian** [comment]
@prury try again thanks

**Comment by prury** [comment]
i think we are good now! thank you @hassnian

**Review by prury — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

lets resolve conflicts @hassnian and can be merged

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 5.16 USD/DOT ~ 9.69 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0x6b9650a306168d2394bf291b3f84a006a07024153ceb43c7a8616dd6753be893](https://polkadot.subscan.io/extrinsic/0x6b9650a306168d2394bf291b3f84a006a07024153ceb43c7a8616dd6753be893)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

File: utils/support.ts [hunk 2]
```diff
@@ -93,13 +103,15 @@ export const canSupport = async (
   api: ApiPromise,
   enabled: boolean,
   multiplyWith = 1,
+  token?: SupportTokens = 'KSM',
```
Comment (preschian) [inline utils/support.ts:106]:
```suggestion
  token: SupportTokens = 'KSM',
```

**Review by preschian — APPROVED** [review]

**Pull request merged**